src/gs-update-monitor.c

```diff
@@ -150,6 +192,269 @@ no_updates_for_a_week (GsUpdateMonitor *monitor)
 	return FALSE;
 }
 
+static void
+app_set_auto_updating (GsApp *app,
+		       gboolean auto_updating)
+{
+	/* we always have to set it to NULL as otherwise GsApp doesn't allow
+	 * the metadata to be overridden */
+	gs_app_set_metadata_variant (app, APP_METADATA_AUTO_UPDATING, NULL);
+
+	/* only set a value if it's TRUE, otherwise it's not needed because not
+	 * having one is the same as having it as false */
+	if (auto_updating) {
+		g_autoptr(GVariant) tmp = g_variant_new_boolean (auto_updating);
+		gs_app_set_metadata_variant (app, APP_METADATA_AUTO_UPDATING, tmp);
+	}
+}
+
+static gboolean
+app_is_auto_updating (GsApp *app)
+{
+	GVariant *tmp = gs_app_get_metadata_variant (app, APP_METADATA_AUTO_UPDATING);
+	if (tmp == NULL)
+		return FALSE;
+	return g_variant_get_boolean (tmp);
+}
+
+static void
+app_update_finished_cb (GObject *source,
+			GAsyncResult *res,
+			UpdateScheduleHelper *helper)
```

Comment:
Nitpick: Misalignment on the arguments?

Reply:
Another nitpick: I find it slightly more readable to put the `GAsyncReadyCallback` function below the `_async` call — so I’d put `app_update_finished_cb()` immediately below the end of `update_app()`. Then you can read through the code for `update_app()`, the `_async()` call`, and then immediately read through the code in the async callback which is next in the control flow.

The downside to that is that it does require you to declare `app_update_finished_cb()` in advance.

Reply:
About the misalignments. This code uses tabs (I know...) so it shows funny depending on the editor. In emacs it looks aligned so I guess I'll believe that.

Reply:
About defining the callback after its use, I agree with you generally, and it's even a recommendation from the [Clean Code book](https://www.goodreads.com/book/show/3735293-clean-code) IIRC. However, for C code and GNOME in particular, the opposite is more common as it skips having to pre-declare the function. So I prefer to leave it like this.